lib/mappers/booking_events_property_ids_mapper.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+# returns structure: { preperty_id => Array[BookingEventsStructure] (sorted by date time of booking) }
+require 'time'
+
+class BookingEventsPropertyIdsMapper
+  def initialize(events_structures)
+    @events_structures = events_structures
+  end
+
+  def map
+    properties_events, booking_events = events_structures.partition { |event| event.data['property_id'] }
+    booking_events_by_booking_id = booking_events.group_by { |event| event.data['booking_id'] }
+    booking_events_structures_by_property_id = properties_events.map do |property_event|
+      { property_event.data['property_id'] => booking_events_by_booking_id[property_event.data['booking_id']] }
+    end
+
+    # it can be a situation that events json is not sorted properly by created at
+    booking_events_structures_by_property_id.reduce({}) do |h, i|
+      h.merge(i) { |_, old, new| [old, new].flatten.sort_by { |event| Time.parse(event.created_at) } }
```

Comment:
```suggestion
      h.merge(i) { |_, old, new| [old, new].flatten.map(&:created_at).sort_by(&Time.method(:parse)) }
```
\- не уверен как лучше :thinking: